### ISSUE TITLE
How do I compile with Fedora?

### ISSUE THREAD

**Issue by cbmeeks** [punesemu/puNES#194]
Trying to build the source I get an error saying autoreconf is not found.
I don't believe that is something Fedora uses (I think it comes with Arch?).

Any tips would be appreciated.

Thanks

**Comment by punesemu** [comment]
At a minimum you need the autotools packages: autoconf automake. But they're not the only ones you'll need, take a look at the packages needed to compile it under ubuntu to get an idea of what you will need.
https://github.com/punesemu/puNES#example-on-how-to-compile-on-ubuntu-2110

**Comment by punesemu** [comment]
Now unfortunately I don't have much time but maybe over the weekend I could create a virtual machine in which to install fedora to understand how to compile it.

**Comment by punesemu** [comment]
These are the steps I was able to successfully compile on a Fedora 35

    $ sudo dnf install git cmake automake autoconf pkg-config libudev-devel alsa-lib-devel mesa-libGLU-devel
    $ sudo dnf install qt5-qtbase-devel qt5-qttools-devel qt5-qtsvg-devel libX11-devel libXrandr-devel

if you want support for the nvidia-cg-toolkit (external shaders):

    $ sudo dnf install https://download1.rpmfusion.org/nonfree/fedora/rpmfusion-nonfree-release-$(rpm -E %fedora).noarch.rpm
    $ sudo dnf update
    $ sudo dnf install Cg

if you want support to record video:

    $ sudo dnf install https://download1.rpmfusion.org/free/fedora/rpmfusion-free-release-$(rpm -E %fedora).noarch.rpm
    $ sudo dnf update
    $ sudo dnf install ffmpeg-devel

If your distribution uses Wayland the emulator may not start.
This is an initial patch that adds support for Wayland and will allow you to start the emulator: [punes_initial_wayland_support.patch](https://www.dropbox.com/s/cl4irqhqyslkmdz/punes_initial_wayland_support.patch?dl=1)

**Comment by punesemu** [comment]
Ok, with this https://github.com/punesemu/puNES/commit/5b69182f19b47734589e842b6c9acc43e001ae7d the emulator should also start under Wayland. Fullscreen still does not work but I will also fix this problem.

**Comment by cbmeeks** [comment]
Man, I don't know why I cannot get this to work.

So I was finally able to configure/make.

When I try to run it, I get this:

    [cbmeeks@fedora puNES]$ ./src/punes 
    QSocketNotifier: Can only be used with threads started with QThread
    INFO: path /home/cbmeeks/.puNES
    OPENGL: Unknown error
    Unable to initiliazed GFX API

I believe I have OpenGL installed:

    glxinfo | grep OpenGL

    OpenGL vendor string: AMD
    OpenGL renderer string: AMD RENOIR (DRM 3.44.0, 5.16.16-200.fc35.x86_64, LLVM 13.0.0)
    OpenGL core profile version string: 4.6 (Core Profile) Mesa 21.3.8
    OpenGL core profile shading language version string: 4.60
    OpenGL core profile context flags: (none)
    OpenGL core profile profile mask: core profile
    OpenGL core profile extensions:
    OpenGL version string: 4.6 (Compatibility Profile) Mesa 21.3.8
    OpenGL shading language version string: 4.60
    OpenGL context flags: (none)
    OpenGL profile mask: compatibility profile
    OpenGL extensions:
    OpenGL ES profile version string: OpenGL ES 3.2 Mesa 21.3.8
    OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
    OpenGL ES profile extensions:

**Comment by punesemu** [comment]
I've already fixed this with https://github.com/punesemu/puNES/commit/5b69182f19b47734589e842b6c9acc43e001ae7d.
You have to update the sources with this command:

    [cbmeeks@fedora puNES]$ git pull

then 

    [cbmeeks@fedora puNES]$ make
    [cbmeeks@fedora puNES]$ ./src/punes

**Comment by cbmeeks** [comment]
Sorry, I missed that part.

That worked!

Thanks!

![image](https://user-images.githubusercontent.com/49341/160149914-2b557512-269a-41e7-8eb1-0413b2a512ce.png)

**Comment by punesemu** [comment]
Just out of curiosity, I have never tried the emulator on Wayland (having never used Wayland) can you tell me how it behaves in terms of performance? Do you notice slowdowns during emulation? I know that the fullscreen still doesn't work (at least it doesn't work in the virtual machine I created) have you noticed any other particular problem?

**Comment by cbmeeks** [comment]
So you're saying I have to fire up the emulator, take time off from household chores, and play some games so that I can report back??

Well, if you insist I will do it.   :-D

LOL

**Comment by cbmeeks** [comment]
So I played Blaster Master for a little bit.  Performance seems good.  Sometimes the FPS would drop to 59 but it was almost always 60 FPS.

I would see occasional slowdown but that would have happened on real hardware.  Still, the FPS was at 60.  So it was emulating real slow down.

I run the window doubled to fit my monitor better and I don't use any GPU filters.  However, I tried them out and they didn't seem to affect performance at all.  I have to admit, the CRT filters looked really good and I don't normally use filters.

For reference, my machine is a Ryzen 7, 8 core (16 thread) and 32 GiB of RAM with the built-in GPU of the Ryzen.
